### ISSUE TITLE
Submissions do not contain references to previous submissions

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#27356]
Test InteractiveSessionTests (being added in https://github.com/dotnet/roslyn/pull/27312) is skipped because the compilation of a non-initial submission has a distinct assembly from its previous submission, contains no references (e.g. `Compilation.References`) to the previous submission, and yet contains symbols that come from the previous submission. Since the `Compilation.IsSymbolAccessibleWithin` API requires that the symbols it is given are either defined in or referenced by the compilation, it is not possible to use that API for interactive scenarios.

**Comment by AlekseyTs** [comment]
> Since the Compilation.IsSymbolAccessibleWithin API requires that the symbols it is given are either defined in or referenced by the compilation, it is not possible to use that API for interactive scenarios.

For the purpose of implementing IsSymbolAccessibleWithin API, we can treat them as implicit references.  Compiler has all the required information.

Of course, we should figure out what is the right way to expose information through the public API, but that shouldn't block implementation of IsSymbolAccessibleWithin API because it can use what is available internally without any limitation.

**Comment by gafter** [comment]
Yes, it is possible for the implementation of IsSymbolAccessibleWithin to compensate for this bug as an alternative to the bug being fixed.

**Comment by AlekseyTs** [comment]
> Yes, it is possible for the implementation of IsSymbolAccessibleWithin to compensate for this bug as an alternative to the bug being fixed.

I don't have any preference regarding how the problem with implementation of IsSymbolAccessibleWithin API is addressed. I think that this issue should not be used as an excuse to ship IsSymbolAccessibleWithin in a broken state.
